test/skimNtuple2016_HHbtag.cpp

```diff
@@ -2668,598 +2581,455 @@ int main (int argc, char** argv)
-	  idAndIsoSF_leg1_deep_vsMu_down[bin] = Deep_antiMu_vloose ->getSFvsEta(tau1eta, tau1Genmatch, "Down");
+	if(isMC and leg2eta < 2.4 and pType == 3) { //MuMu
+	  idSF_leg2 = myIDandISOScaleFactor[0]->get_ScaleFactor(leg2pt, leg2eta)*myIDandISOScaleFactor[2]->get_ScaleFactor(leg2pt, leg2eta);
 	}
-      }
-
-      for (int bin = 0; bin < (int) isthisSDet_IDbin_first.size(); bin++)
-      {
-	if (isthisSDet_IDbin_first[bin])
-	{
-	  idAndIsoSF_leg1_deep_vsEle_up[bin]   = Deep_antiEle_vvloose ->getSFvsEta(tau1eta, tau1Genmatch,   "Up");
-	  idAndIsoSF_leg1_deep_vsEle_down[bin] = Deep_antiEle_vvloose ->getSFvsEta(tau1eta, tau1Genmatch, "Down");
+	else if(isMC and leg2eta < 2.4 and pType == 4) { //EleEle
+	  idSF_leg2 = myIDandISOScaleFactor[1]->get_direct_ScaleFactor(leg2pt, leg2eta);
 	}
-      }
-
-      // Up/down variations Leg 2
-      vector<float> idAndIsoSF_leg2_deep_vsJet_pt_up (5, idAndIsoSF_leg2_deep_vsJet_pt); // in bins of pt: 20, 25, 30, 35, 40, infty
-      vector<float> idAndIsoSF_leg2_deep_vsEle_up    (2, idAndIsoSF_leg2_deep_vsEle);    // in bins of eta: barrel, endcap
-      vector<float> idAndIsoSF_leg2_deep_vsMu_up     (5, idAndIsoSF_leg2_deep_vsMu);     // in bins of eta, edges at 0, 0.4, 0.8, 1.2, 1.7, infty
-
-      vector<float> idAndIsoSF_leg2_deep_vsJet_pt_down (5, idAndIsoSF_leg2_deep_vsJet_pt); // in bins of pt: 20, 25, 30, 35, 40, infty
-      vector<float> idAndIsoSF_leg2_deep_vsEle_down    (2, idAndIsoSF_leg2_deep_vsEle);    // in bins of eta: barrel, endcap
-      vector<float> idAndIsoSF_leg2_deep_vsMu_down     (5, idAndIsoSF_leg2_deep_vsMu);     // in bins of eta, edges at 0, 0.4, 0.8, 1.2, 1.7, infty
-
-      for (int bin = 0; bin < (int) isthisPt_IDbin_second.size(); bin++)
-      {
-	if (isthisPt_IDbin_second[bin])
-	{
-	  idAndIsoSF_leg2_deep_vsJet_pt_up[bin]   = Deep_antiJet_medium_pt ->getSFvsPT(tau2pt, tau2Genmatch,   "Up");
-	  idAndIsoSF_leg2_deep_vsJet_pt_down[bin] = Deep_antiJet_medium_pt ->getSFvsPT(tau2pt, tau2Genmatch, "Down");
 
-	  //Additional uncertainty see https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauIDRecommendationForRun2#Corrections_to_be_applied_to_gen
-
-	  double error_up   = fabs(idAndIsoSF_leg2_deep_vsJet_pt - idAndIsoSF_leg2_deep_vsJet_pt_up[bin]);
-	  double error_down = fabs(idAndIsoSF_leg2_deep_vsJet_pt - idAndIsoSF_leg2_deep_vsJet_pt_down[bin]);
-
-	  if (tau2pt < 100)
+	float except_VsJet, except_vsMu, except_vsEle;
+	// EleTau and MuTau
+	if (isMC and (pType == 0 or pType == 1))
 	  {
-	    idAndIsoSF_leg2_deep_vsJet_pt_up[bin]   = idAndIsoSF_leg2_deep_vsJet_pt*1.03 + error_up;
-	    idAndIsoSF_leg2_deep_vsJet_pt_down[bin] = idAndIsoSF_leg2_deep_vsJet_pt*0.97 - error_down;
+		idSF_deep_dm	 = idSF_leg1 * idSF_leg2_deep_vsJet_dm;
+		idSF_deep_pt	 = idSF_leg1 * idSF_leg2_deep_vsJet_pt;
+		idSF_deep_2d	 = idSF_leg1 * idSF_leg2_deep_vsJet_2d;
+		fakeRateSF_deep	 = idSF_leg2_deep_vsEle * idSF_leg2_deep_vsMu;
+
+		except_VsJet = idSF_leg1 * idSF_leg2_deep_vsEle * idSF_leg2_deep_vsMu;
```

Comment:
I think I understand after a couple more readings. This a very smart trick to have the proper combinations of central values and systematic variations, right?

Reply:
"Very smart" is quite an exaggeration :)
It is just a variable that stores all values required to calculate the weights *except* one of them. And yes, it is useful for the systematics, to avoid computing the same quantities over and over again.